File: Brute Force/Depth-First Search/tree.cpp
#include "algorithm-visualizer.h"
#include <iostream>
#include <string>

GraphTracer tracer("GraphTracer");
LogTracer logger("LogTracer");

int G[11][11] = { // G[i][j] indicates whether the path from the i-th node to the j-th node exists or not
	{0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
};

void InitDepthFirstSearch_tree()
{
	tracer.log(logger);
	Layout::setRoot(VerticalLayout({ tracer,logger }));
	tracer.set(G);
	tracer.layoutTree(0);
	Tracer::delay();
}
void DFS(int node, int parent = NULL)
{
	tracer.visit(node, parent);
	Tracer::delay();
	for (int i = 0; i < 11; i++) {
		if (G[node][i])
			DFS(i, node);
	}
}

int main()
{
	InitDepthFirstSearch_tree();
	DFS(0);
}

File: Brute Force/Depth-First Search/tree.java
import org.algorithm_visualizer.*;
import java.util.*;

public class Main {
	//visualizer{
	private static GraphTracer tracer;
	private static LogTracer logger;
	//}
	private static Object[][] graph;
	private static String[] visited;
	
	Main() 
	{                                     
		int[][] G =	{ 
				{0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
				{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
		};
		
        //visualizer{
		tracer = new GraphTracer().directed(true).weighted(false);
		logger = new LogTracer();
		Layout.setRoot(new VerticalLayout(new Commander[] {tracer, logger}));
		tracer.log(logger);
		tracer.set(G);
		tracer.layoutTree(0);
		Tracer.delay();
        //}
		DFS(G,0,0);
	}

	void DFS(int[][] G, int node, int parent) { 
		//visualizer{
		tracer.visit(node, parent);
		Tracer.delay();
		//}
		for (int i = 0; i < G[node].length; i++) {
			if (G[node][i] != 0) { 
				DFS(G, i, node); 
			}
		}
	}

	public static void main(String[] args) {
		new Main();
	}
}